website/server/models/group.js

```diff
@@ -392,7 +392,18 @@ schema.methods.sendChat = function sendChat (message, user) {
   let newMessage = chatDefaults(message, user);
 
   this.chat.unshift(newMessage);
-  this.chat.splice(200);
+
+  const MAX_CHAT_COUNT = 200;
+  const MAX_SUBBED_GROUP_CHAT_COUNT = 400;
+
+  let maxCount = MAX_CHAT_COUNT;
+
+  let plan = this.purchased.plan;
+  if (plan && plan.customerId && !plan.dateTerminated) {
```

Comment:
It is definitely possible to have a value for purchased.plan.dateTerminated in the past - that indicates an inactive subscription. If there's any checks for active user subscriptions that rely on dateTerminated not existing, that would be a bug.

A dateTerminated in the past could also indicate a newly-created subscription - one that has just been made one or more days after an old one terminated. When the new one is created,  dateTerminated should be set to null, but I'm pretty sure there's still bug(s) that prevent that happening.